### PR TITLE
[SPARK-40214][PYTHON][SQL] add 'get' to functions

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
expose `get` to dataframe functions

### Why are the changes needed?
for function parity

### Does this PR introduce _any_ user-facing change?
yes, new API

### How was this patch tested?
added UT

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -3958,6 +3958,26 @@ object functions {
...
+  }
+
+  /**
+   * Returns element of array at given (0-based) index. If the index points
+   * outside of the array boundaries, then this function returns NULL.
+   *
+   * @group collection_funcs
+   * @since 3.4.0
+   */
+  def get(column: Column, index: Int): Column = get(column, lit(index))
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:3979]: Let's probably don't add this (see the comments on the top of this file)

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -4845,6 +4849,73 @@ def element_at(col: "ColumnOrName", extraction: Any) -> Column:
...
+    -----
+    The position is not 1 based, but 0 based index.
+
+    See Also
+    --------
+    :meth:`element_at`
+
+    Examples
+    --------
+    >>> from pyspark.sql.functions import col, get
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:4877]: Let's remove this line since we're in this module.

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Comment by zhengruifeng** [comment]
Merged to master, thanks @HyukjinKwon for reivew

**Pull request closed without merging**